### ISSUE TITLE
New Feature Request - Accept Batch

### ISSUE THREAD

**Issue by inferno46n2** [nullquant/ComfyUI-BrushNet#62]
Hello,

Could you kindly make this accept batch? It would make it much more compatible with a lot of other tools.

**Comment by nullquant** [comment]
Batch over what? Could you post a scheme of proposed workflow? So I would know what kind of inputs BrushNode will get and from what nodes.

**Comment by inferno46n2** [comment]
Batches of images. Basically if I have a folder with images, or a video, I want to be able to pipe this straight into your nodes such that it processes the images as a batch. It would be the node pack below.

Right now your nodes do not accept batches and only work on single images. Actually they do accept batches, just does not actually function properly (likely because the code already exists just not configured properly)

Batches is the first step to getting this working with more things... like video (AnimateDiff),, but also is a massive quality of life improvment.

https://github.com/nullquant/ComfyUI-BrushNet/issues/62#issuecomment-2095266517

**Comment by nullquant** [comment]
Thank you, I will look into it.

**Comment by nullquant** [comment]
I've done it. Let's try!

**Comment by inferno46n2** [comment]
> I've done it. Let's try!

Not working unfortunatly. 6 Image batch input, but only 1 image coming out of the ksampler (see image below for workflow)

FETCH DATA from: D:\COMFYUI\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
[comfy_mtb] | WARNING -> Found multiple match, we will pick the first D:\COMFYUI\ComfyUI\models\upscale_models
['D:\\COMFYUI\\ComfyUI\\models\\upscale_models', 'C:\\Users\\infer\\OneDrive\\Documents\\Auto1111\\stable-diffusion-webui\\models/ESRGAN', 'C:\\Users\\infer\\OneDrive\\Documents\\Auto1111\\stable-diffusion-webui\\models/RealESRGAN', 'C:\\Users\\infer\\OneDrive\\Documents\\Auto1111\\stable-diffusion-webui\\models/SwinIR']
got prompt
[rgthree] Using rgthree's optimized recursive execution.
[rgthree] First run patching recursive_output_delete_if_changed and recursive_will_execute.
[rgthree] Note: If execution seems broken due to forward ComfyUI changes, you can disable the optimization from rgthree settings in ComfyUI.
model_type EPS
Using xformers attention in VAE
Using xformers attention in VAE
BrushNet model type: SD1.5
BrushNet model file: D:\COMFYUI\ComfyUI\models\inpaint\random_mask_brushnet_ckpt\diffusion_pytorch_model.safetensors
BrushNet SD1.5 model is loaded
Requested to load SD1ClipModel
Loading 1 new model
D:\COMFYUI\ComfyUI\comfy\ldm\modules\attention.py:345: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:263.)
  out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)
Base model type: SD1.5
BrushNet image.shape = torch.Size([1, 1080, 1920, 3]) mask.shape = torch.Size([1, 1080, 1920])
Requested to load AutoencoderKL
Loading 1 new model
BrushNet CL: image_latents shape = torch.Size([2, 4, 135, 240]) interpolated_mask shape = torch.Size([2, 1, 135, 240])
Requested to load BaseModel
Loading 1 new model
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: do_classifier_free_guidance is True
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([2, 5, 135, 240])
D:\COMFYUI\ComfyUI\venv\lib\site-packages\diffusers\models\resnet.py:328: FutureWarning: `scale` is deprecated and will be removed in version 1.0.0. The `scale` argument is deprecated and will be ignored. Please remove it, as passing it will raise an error in the future. `scale` should directly be passed while calling the underlying pipeline component i.e., via `cross_attention_kwargs`.
  deprecate("scale", "1.0.0", deprecation_message)
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:07<00:00,  2.58it/s]
Prompt executed in 15.75 seconds
got prompt
[rgthree] Using rgthree's optimized recursive execution.
Base model type: SD1.5
BrushNet image.shape = torch.Size([1, 1080, 1920, 3]) mask.shape = torch.Size([1, 1080, 1920])
BrushNet CL: image_latents shape = torch.Size([2, 4, 135, 240]) interpolated_mask shape = torch.Size([2, 1, 135, 240])
Requested to load BaseModel
Loading 1 new model
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: do_classifier_free_guidance is True
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([2, 5, 135, 240])
D:\COMFYUI\ComfyUI\venv\lib\site-packages\diffusers\models\resnet.py:328: FutureWarning: `scale` is deprecated and will be removed in version 1.0.0. The `scale` argument is deprecated and will be ignored. Please remove it, as passing it will raise an error in the future. `scale` should directly be passed while calling the underlying pipeline component i.e., via `cross_attention_kwargs`.
  deprecate("scale", "1.0.0", deprecation_message)
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:07<00:00,  2.63it/s]
Prompt executed in 9.35 seconds
got prompt
[rgthree] Using rgthree's optimized recursive execution.
Base model type: SD1.5
BrushNet image.shape = torch.Size([1, 1080, 1920, 3]) mask.shape = torch.Size([1, 1080, 1920])
BrushNet CL: image_latents shape = torch.Size([2, 4, 135, 240]) interpolated_mask shape = torch.Size([2, 1, 135, 240])
Requested to load BaseModel
Loading 1 new model
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: do_classifier_free_guidance is True
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([2, 5, 135, 240])
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:07<00:00,  2.66it/s]
Prompt executed in 9.15 seconds
got prompt
[rgthree] Using rgthree's optimized recursive execution.
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: do_classifier_free_guidance is True
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([2, 5, 135, 240])
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:09<00:00,  2.18it/s]
Prompt executed in 9.96 seconds
got prompt
[rgthree] Using rgthree's optimized recursive execution.
Base model type: SD1.5
BrushNet image.shape = torch.Size([2, 1080, 1920, 3]) mask.shape = torch.Size([2, 1080, 1920])
BrushNet CL: image_latents shape = torch.Size([4, 4, 135, 240]) interpolated_mask shape = torch.Size([4, 1, 135, 240])
Requested to load BaseModel
Loading 1 new model
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: do_classifier_free_guidance is True
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([4, 5, 135, 240])
BrushNet inference: sample batch is less, than CL!
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([2, 5, 135, 240])
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:09<00:00,  2.19it/s]
Prompt executed in 12.02 seconds
got prompt
[rgthree] Using rgthree's optimized recursive execution.
Base model type: SD1.5
BrushNet image.shape = torch.Size([6, 1080, 1920, 3]) mask.shape = torch.Size([6, 1080, 1920])
BrushNet CL: image_latents shape = torch.Size([12, 4, 135, 240]) interpolated_mask shape = torch.Size([12, 1, 135, 240])
Requested to load BaseModel
Loading 1 new model
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: do_classifier_free_guidance is True
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([12, 5, 135, 240])
BrushNet inference: sample batch is less, than CL!
BrushNet inference: sample torch.Size([2, 4, 135, 240]) , CL torch.Size([2, 5, 135, 240])
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:08<00:00,  2.41it/s]
Prompt executed in 14.34 seconds


![image](https://github.com/nullquant/ComfyUI-BrushNet/assets/73665023/e367b38f-2be4-4b8f-ac41-6b6874fcde5b)

**Comment by nullquant** [comment]
I updated the code, try it.

**Comment by inferno46n2** [comment]
> I updated the code, try it.

nope

Base model type: SD1.5
BrushNet image.shape = torch.Size([3, 1080, 1920, 3]) mask.shape = torch.Size([3, 1080, 1920])
BrushNet CL: image_latents shape = torch.Size([6, 4, 135, 240]) interpolated_mask shape = torch.Size([6, 1, 135, 240])
Requested to load BaseModel
Loading 1 new model
  0%|                                                                                           | 0/20 [00:00<?, ?it/s]BrushNet inference: sample torch.Size([1, 4, 135, 240]) , CL torch.Size([3, 5, 135, 240])
BrushNet inference: sample batch is less, than CL!
BrushNet inference: sample torch.Size([1, 4, 135, 240]) , CL torch.Size([1, 5, 135, 240])
BrushNet inference: sample torch.Size([1, 4, 135, 240]) , CL torch.Size([3, 5, 135, 240])
BrushNet inference: sample batch is less, than CL!
BrushNet inference: sample torch.Size([1, 4, 135, 240]) , CL torch.Size([1, 5, 135, 240])
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:09<00:00,  2.18it/s]
Prompt executed in 12.96 seconds

**Comment by nullquant** [comment]
Sorry for inconvenience, found a bug, no, a BUG in codebase. It should work now.
